### PR TITLE
ETK: Fix the start page options modal is still visible

### PR DESCRIPTION
**Pull request by arthur791004**
Related to https://github.com/Automattic/wp-calypso/pull/83750, p1702774182812059-slack-C048CUFRGFQ

## Proposed Changes

* Fix the start page options modal is still visible

## Testing Instructions

* Apply this change to your sandbox
* Sandbox the public-api and your site
* Activate the TT4 theme on your site
* Create a new page
* Ensure the start page options modal from Core doesn't show
* Ensure the page pattern shows in the starter page template modal

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: apps/editing-toolkit/editing-toolkit-plugin/block-patterns/class-block-patterns-from-api.php [hunk 1]
```diff
@@ -317,8 +317,8 @@ private function update_pattern_block_types() {
 			if ( $post_content_offset !== false ) {
 				unregister_block_pattern( $pattern['name'] );
 
-				$pattern['blockTypes'] = array_splice( $pattern['blockTypes'], $post_content_offset, 1 );
-				$pattern_name          = $pattern['name'];
+				array_splice( $pattern['blockTypes'], $post_content_offset, 1 );
```
Comment (arthur791004) [inline apps/editing-toolkit/editing-toolkit-plugin/block-patterns/class-block-patterns-from-api.php:320]: The `array_splice` returns an array consisting of the extracted elements... I'm unsure why it worked before 🙈

**Comment by taipeicoder** [comment]
> Ensure the created page pattern shows in the starter page template modal
 
@arthur791004  I followed the instructions in #83750 to create a page pattern under `/twentytwentyfour/patterns` but it didn't show up in the start page options modal. Am I missing anything?

`/twentytwentyfour/patterns/page-example.php`
```
/**
 * Title: Example Theme Page Pattern
 * Slug: twentytwentyfour/example-theme-page-pattern
 * Categories: page, featured
 * Keywords: page pattern
 * Block Types: core/post-content
 * Post Types: page, wp_template
 */
?>

<!-- wp:group {"layout":{"type":"constrained"}} -->
<div class="wp-block-group">
   <!-- wp:paragraph -->
  <p>Example Theme Page Pattern</p>
  <!-- /wp:paragraph -->
</div>
<!-- /wp:group -->
```

**Review by fushar — APPROVED** [review]
Working well.

> @arthur791004 I followed the instructions in https://github.com/Automattic/wp-calypso/pull/83750 to create a page pattern under /twentytwentyfour/patterns but it didn't show up in the start page options modal. Am I missing anything?

@taipeicoder I noticed this as well, but then I found that somehow new page patterns / updates to existing page patterns are only visible if we change the active theme to another theme first and then activate TT4 back... not sure if this is intended or not 😅

**Comment by taipeicoder** [comment]
> Working well.
> 
> > @arthur791004 I followed the instructions in #83750 to create a page pattern under /twentytwentyfour/patterns but it didn't show up in the start page options modal. Am I missing anything?
> 
> @taipeicoder I noticed this as well, but then I found that somehow new page patterns / updates to existing page patterns are only visible if we change the active theme to another theme first and then activate TT4 back... not sure if this is intended or not 😅

@fushar that worked (somehow), thank you!

https://github.com/Automattic/wp-calypso/assets/797888/09d817ea-828d-476a-9abc-8c67ec5e1582

**Review by taipeicoder — APPROVED** [review]

**Comment by arthur791004** [comment]
Thank you 👍

**Pull request merged**